feat: try to find actual target directory to place wasi deps into

Gentle opened the pull request:
Scenario: in a workspace setup, the library extracts the wasi-deps to every crate that links against libpython, this code tries to instead find the actually used cargo target directory

The code searches for the last occurrence of `/wasm32-wasi/` in the OUT_DIR variable so this works also for setups where the user set the CARGO_TARGET_DIR environment variable

Example:

workspace/target <= I want the files to be put here
workspace/library_a/target/wasm32-wasi/wasi-deps <= these three are currently being created
workspace/library_b/target/wasm32-wasi/wasi-deps <= these three are currently being created
workspace/library_c/target/wasm32-wasi/wasi-deps <= these three are currently being created

ereslibre commented on python/tools/wlr-libpy/src/bld_cfg.rs:
```diff
@@ -15,20 +15,32 @@ const LIBPYTHON_URL: &str = "https://github.com/vmware-labs/webassembly-language
 
-    let wasi_deps_path = Path::new(WASI_DEPS_PATH);
+    let out_dir = if let Ok(mut out_dir) = std::env::var("OUT_DIR") {
+        if let Some(position) = out_dir.rfind("/wasm32-wasi/") {
```
I think we should convert the OUT_DIR to a `PathBuf`, and use `std::path::Path::components` to find out. This will be more solid if the trailing slash is missing.

ereslibre commented on python/tools/wlr-libpy/src/bld_cfg.rs:
```diff
@@ -15,20 +15,32 @@ const LIBPYTHON_URL: &str = "https://github.com/vmware-labs/webassembly-language
-    let wasi_deps_path = Path::new(WASI_DEPS_PATH);
+    let out_dir = if let Ok(mut out_dir) = std::env::var("OUT_DIR") {
+        if let Some(position) = out_dir.rfind("/wasm32-wasi/") {
+            out_dir.truncate(position + 13);
```
We should do this logic with `PathBuf`, having constants like this is very brittle and error prone to maintain.

ereslibre reviewed (changes requested):
Thank you for your contribution @Gentle! I think this is a nice improvement, but I added some comments to make the code a bit more maintainable.

Angelmmiguel commented:
No worries @Gentle, just ping us when the changes are ready. Thank you for your contribution :)

Gentle committed:
search for target dir

resolution order:

- CARGO_TARGET_DIR variable
- the first parent directory that has a Cargo.lock
- if both were not successful, fallback to the previous behavior

Gentle commented:
I rewrote the patch, it now searches for the correct target directory in this order:

- CARGO_TARGET_DIR variable
- the first parent directory that has a Cargo.lock
- if both were not successful, fallback to the previous behavior

ereslibre commented:
Hello @Gentle,

Sorry for the long time to reply and thank you for your contribution. After some long discussions, this project is currently in the process of being archived. However, the development of this project will continue as an independent community fork present at https://github.com/webassemblylabs/webassembly-language-runtimes.

We would be grateful if you could recreate your contribution under that fork and we will work on applying it.

Gentle commented:
thank you, I will move the PR
